src/buf.c

```diff
@@ -266,4 +285,10 @@ struct Buf *buf_append (struct Buf *buf, const void *ptr, int n_elem)
 	return buf;
 }
 
+struct Buf *buf_append_alloc (struct Buf *buf, const void *ptr, int n_elem)
+{
+	buf_append(&alloc_buf, ptr, n_elem);
```

Comment:
Alternately, we could strdup() every string that we put in the buffer, but I'm not so sure that the resulting code will be much better that the solution I've presented.